articles/node-vs-python/index.md

```diff
@@ -0,0 +1,179 @@
+- [Express](https://expressjs.com/) - Unopinionated web framework for Node.js.
+- [Adonis](https://adonisjs.com/) - Full-Stack MVC Framework.
+- [Koa](https://koajs.com/) - Robust Foundation for Web Applications.
+
+#### Package Management
+Python uses a package manager called `pip`. It is extremely easy to use and has a ton of community-published packages. These packages can be found at the [Python Package Index](https://pypi.org/).
+
+It uses a file called **requirements.txt** to keep track of packages in a project.
+
+To install a package in python, just enter:
+
+```
+pip install <package-name>
+```
+
+Node.js uses a package manager called `npm`. The [npm registry](https://www.npmjs.com/) contains open source packages for Node.js, front-end web development, mobile apps and more.
+
+To keep track of packages in a project, Node.js uses a file called **package.json**
+
+To install a Node.js package, just type:
+
+```
+npm i <package-name>
+```
+
+### APIs in Node.js
+
+Let us dive into creating an API in Node.js our API has an endpoint _/read_ which reads data from a table, _sample_ in a MySQL Database.
+
+```js
+/* Importing Libraries */
+
+const mysql = require("mysql");
+const express = require("express");
+const bodyParser = require("body-parser");
+
+/* Initialize Express */
+
+const app = express();
+
+/* Body Parser */
+
+app.use(bodyParser.json());
+
+app.use(bodyParser.urlencoded({ extended: true }));
+
+/* MySQL Config */
+
+const connection = mysql.createConnection({
+  host: "<hostname>",
+  user: "<username>",
+  password: "<password>",
+  database: "<dbname>",
+});
+
+/* Initialize MySQL Connection */
+
+connection.connect((err) => {
+  if (err) throw err;
+  console.log("Successfully connected to the database.");
+});
+
+/* Define API Endpoint */
+
+app.get("/read", (req, res) => {
+  connection.query("SELECT * FROM sample", (err, res, fields) => {
+    if (err) throw err;
+    console.log(res);
+  });
+});
+```
+
+In the snippet above, we initialize a MySQL connection using the Node.js MySQL library. Whenever a get request is sent to the _/read_ endpoint, the database is read.
+
+### APIs in Python
+Let us build a similar API that reads a database (SQLAlchemy) using Python.
+
+```python
+from flask import Flask, request
+from flask_restful import Resource, Api
+from sqlalchemy import create_engine
+from flask.ext.jsonpify import jsonify
+from json import dumps
+
+db = create_engine('mysql://username:password@host/database')
+app = Flask(__name__)
+api = Api(app)
+
+class Read(Resource):
+    def get(self):
+        conn = db.connect() # Connect to Database
+        query = conn.execute("select * from sample") # Query and return json result
+        return jsonify(query)
+
+api.add_resource(Read, '/read')
+
+if __name__ == '__main__':
+     app.run(port='5002')
+```
+
+In the snippet above, we initialize a MySQL connection using the SQLAlchemy library in Python. The Flask library is used for creating API endpoints.
```

Comment:
Your explanation of the code is lacking. You should provide comments in the code that better explain what the various lines of code are for.